var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -181,3 +181,24 @@ def fix_darwin_install(self):
                 join_path(self.prefix.lib, "libpapi.dylib"),
             )
             fs.fix_darwin_install_name(self.prefix.lib)
+
+    test_src_dir = "src/smoke_tests"
+
+    @run_after("install")
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        if os.path.exists(self.test_src_dir):
+            self.cache_extra_test_sources([self.test_src_dir])
+
+    def test(self):
+        test_dir = join_path(self.test_suite.current_test_cache_dir, self.test_src_dir)
+        if not os.path.exists(test_dir):
+            print("Skipping smoke tests, directory doesn't exist")
```

Comment:
You probably want to raise the new `SkipTest` exception, e.g., `raise SkipTest("Tests not supported for this version")` and remove the `return`.